arkouda/strings.py

```diff
@@ -1121,10 +1121,19 @@ def flatten(
             return self.split(delimiter, return_segments=return_segments)
         else:
             cmd = "segmentedFlatten"
-            args = "{} {} {} {} {}".format(
-                self.entry.name, self.objtype, return_segments, regex, json.dumps([delimiter])
+            repMsg = cast(
+                str,
+                generic_msg(
+                    cmd=cmd,
+                    args={
+                        "values": self.entry,
+                        "objtype": self.objtype,
+                        "return_segs": return_segments,
+                        "regex": regex,
+                        "delim": [delimiter],
```

Comment:
did a deep dive with git blame on why we were doing a JSON dumps on `[delimiter]`. I thought I did this but it predates the regex code and has been like this at least two years

I think that was probably because the delimiter can contain spaces which would break the old message passing. I *think* with the new message passing, we can just do `"delim": delimiter` here since everything is json

Reply:
I was just matching the existing. Updated this.